src/components/Sidebar/SidebarAuth/index.tsx

```diff
@@ -85,23 +101,56 @@ export const SidebarAuth = ({
     clearFields();
   };
 
-  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+  const handleInputChange = async (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
     const { name, value } = event.target;
+    if (name === 'username') {
+      // make a throttled request to the api to check if the username is available
+      // if not, show a message
+      const { patterns } = validateUsername(value);
+      setUsernamePatterns(patterns);
+      await checkIfUsernameIsNotTaken(
+        value,
+        setNotification,
+        setUsernamePatterns
+      );
+    }
     setNotification(null);
```

Comment:
This line is confusing me a tiny bit. Do I understand it correctly, that the purpose of this line is to reset the notification after each input for each field? If yes, i'd suggest to put it _before_ the if block `if(name === 'username') {...}`. What confuses me is that right now it reads like we might set a notification in that if block, and just afterwards, we set the notification it to `null`.

Reply:
Since the checkIfUsernameIsTaken is async it will be executed after the line 119. There is a HTTP request to the API involved in that functoin